sdk/src/wallet/migration/migrate_4.rs

```diff
@@ -0,0 +1,260 @@
+        migrate_nodes(nodes)?;
+    }
+    Ok(())
+}
+
+fn migrate_nodes(nodes: &mut serde_json::Value) -> Result<()> {
+    for node in nodes
+        .as_array_mut()
+        .ok_or(Error::Storage("malformatted nodes".to_owned()))?
+    {
+        if let Some(node) = node.as_object_mut() {
+            check_omitted_opt("auth", node);
+        }
+    }
+    Ok(())
+}
+
+fn migrate_account(account: &mut serde_json::Value) -> Result<()> {
+    for output_data in account["outputs"]
+        .as_object_mut()
+        .ok_or(Error::Storage("malformatted outputs".to_owned()))?
+        .values_mut()
+    {
+        migrate_output(&mut output_data["output"])?;
+    }
+
+    for output_data in account["unspentOutputs"]
+        .as_object_mut()
+        .ok_or(Error::Storage("malformatted unspent outputs".to_owned()))?
+        .values_mut()
+    {
+        migrate_output(&mut output_data["output"])?;
+    }
+
+    for transaction in account["transactions"]
+        .as_object_mut()
+        .ok_or(Error::Storage("malformatted transactions".to_owned()))?
+        .values_mut()
+    {
+        migrate_transaction(transaction)?;
+    }
+
+    for transaction in account["incomingTransactions"]
+        .as_object_mut()
+        .ok_or(Error::Storage("malformatted incoming transactions".to_owned()))?
+        .values_mut()
+    {
+        migrate_transaction(transaction)?;
+    }
+
+    if let Some(foundries) = account.get_mut("nativeTokenFoundries") {
+        for foundry_output in foundries
+            .as_object_mut()
+            .ok_or(Error::Storage("malformatted foundry outputs".to_owned()))?
+            .values_mut()
+        {
+            migrate_output(foundry_output)?;
+        }
+    }
+
+    Ok(())
+}
+
+fn migrate_output(output: &mut serde_json::Value) -> Result<()> {
+    let output = output
+        .as_object_mut()
+        .ok_or(Error::Storage("malformatted output".to_owned()))?;
+    check_omitted_str("stateMetadata", output, "0x");
+    check_omitted_list("features", output);
+    check_omitted_list("immutableFeatures", output);
+    if let Some(features) = output.get_mut("features") {
+        for feature in features
+            .as_array_mut()
+            .ok_or(Error::Storage("malformatted features".to_owned()))?
+        {
+            migrate_feature(feature)?;
+        }
+    }
+    if let Some(features) = output.get_mut("immutableFeatures") {
+        for feature in features
+            .as_array_mut()
+            .ok_or(Error::Storage("malformatted immutable features".to_owned()))?
+        {
+            migrate_feature(feature)?;
+        }
+    }
+    Ok(())
+}
+
+fn migrate_transaction(transaction: &mut serde_json::Value) -> Result<()> {
+    let transaction = transaction
+        .as_object_mut()
+        .ok_or(Error::Storage("malformatted transaction".to_owned()))?;
+    check_omitted_str("note", transaction, "");
+    check_omitted_str("blockId", transaction, "");
```

Comment:
Shouldn't it be "0x"?

Reply:
Oh but it'll go in the null check, not the string check, I guess

Reply:
Actually it should be the other method for `Options` since blockId can't be empty. Changed it.